action.yml

```diff
@@ -29,35 +34,92 @@ outputs:
 runs:
   using: composite
   steps:
+    - id: resolveimages
+      if: ${{ inputs.registry-configuration != '' }}
+      name: Resolve source image
+      uses: docker://public.ecr.aws/l7o7z1g8/services/registry-config:0.0.22
+      with:
+        entrypoint: /registry-config
+        args: --config="${{ inputs.registry-configuration }}" resolve "${{ inputs.src }}" resolved-image.file
+
     - id: cranecopy
       name: Copy a remote image from a given source to a destination
       uses: docker://gcr.io/go-containerregistry/crane:debug
       run: |
-        set -euo pipefail
+        set -uo pipefail
+        
+        dest_image=${{ inputs.destination }}
+        if [[ "${dest_image}" == *@sha256:* ]]; then
+          echo "The destination image reference is invalid: contains a digest"
+          exit 1
+        fi
 
-        if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
-          echo "Validating source image ${{ inputs.src }}"
-          crane validate --remote ${{ inputs.src }} --platform ${{ inputs.platform }}
+        if [ ! -e resolved-image.file ]; then
+          echo '${{ inputs.src }}' > resolved-image.file
+        fi
+
+        set +e
+        
+        image_with_digest=
+        while IFS= read -r line; do
+          if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
+            echo "Validating source image ${line}"
+            crane validate --remote ${line} --platform ${{ inputs.platform }}
+
+            if [ $? -ne 0 ]; then
+              echo "Failed to validate image: $line"
+              continue
+            fi
+          fi
+
+          echo "Getting image digest of source image ${line}"
+          image_with_digest=$(crane digest ${line} --full-ref --platform ${{ inputs.platform }})
+        
+          if [ $? -ne 0 ]; then
+            echo "Failed get image digest: $line"
+            image_with_digest=
+            continue
+          fi
+
+          echo "Found source image to copy: ${image_with_digest}"
+
+          echo "Copying source image ${image_with_digest} to destination ${{ inputs.destination }}"
+          crane copy ${image_with_digest} ${{ inputs.destination }} --platform ${{ inputs.platform }}
+
+          if [ $? -ne 0 ]; then
+            echo "Failed to copy image: $line"
+            image_with_digest=
+            continue
+          fi
+          break
+        done < "resolved-image.file"
+        
+        rm resolved-image.file
```

Comment:
we should remove the file after the `set -e` to capture any (inadvertent) issues encountered in removing the file.

Reply:
Moved